it/src/main/scala/org/zalando/nakadi/client/scala/ClientFactory.scala

```diff
@@ -3,37 +3,71 @@ package org.zalando.nakadi.client.scala
 import org.zalando.nakadi.client.utils.ClientBuilder
 import java.util.function.Supplier
 
+import scala.util.Properties
+
 object ClientFactory {
-  import sys.process._
   import scala.language.postfixOps
+
+
+  val ENV_NAKADI_HOST = "NAKADI_HOST"
+  val ENV_NAKADI_PORT = "NAKADI_PORT"
+  val ENV_OAUTH2_TOKEN = "OAUTH2_TOKEN"
+
+  /**
+   * Possible values are (case-insensitive):
+   * - SANDBOX
+   * - STAGING
+   * - LOCAL
+   */
+  val ENV_NAKDI_KLIENTS_IT_ENVIRONMENT = "NAKADI_KLIENTS_IT_ENVIRONMENT"
+
+
   def OAuth2Token(): Option[() => String] =
-    Option(() => "*")
-  def getJavaClient() =
-    builder().buildJavaClient();
+    Option(() => Properties.envOrElse(ENV_OAUTH2_TOKEN ,"*"))
+
+
+  def getJavaClient() = builder().buildJavaClient();
+
 
   def getScalaClient() = builder().build()
 
-  private def builder() = {
-    //    useSandbox()
-    //        useStaging()
-    useLocal()
+
+  private def builder() = stagingFromDev() match {
+      case Some(staging) =>
+        if(staging == "SANDBOX") useSandbox()
+        else if(staging == "STAGING") useStaging()
+        else useLocal() // LOCAL is always used as fallback
+      case None => {
+        useLocal()
+      }
   }
+
+
+  private def stagingFromDev() =
+    Properties.envOrNone(ENV_NAKDI_KLIENTS_IT_ENVIRONMENT)
+      .map(staging => staging.trim.toUpperCase)
+
+
   private def useLocal() = {
     new ClientBuilder() //
-      .withHost("localhost") //
-      .withPort(8080) //
+      .withHost(Properties.envOrElse(ENV_NAKADI_HOST, "localhost")) //
```

Comment:
Please use the Typesafe Config library in order to harmonize all configuration operations, you can achieve the same with this library. It would also be great to have all configurations in a Central place, perhaps also in **org.zalando.nakadi.client.utils.Conf**